### PR TITLE
[PyAMQP] Fix buff prod async not closing down on stress tests

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_buffered_producer/_buffered_producer_dispatcher_async.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 from __future__ import annotations
-import asyncio
+import asyncio # pylint: disable=unused-import
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_buffered_producer/_buffered_producer_dispatcher_async.py:6]: can we remove this if it's not being used?

**Review by annatisch — COMMENTED** [review]
Changes look good. We probably should be doing the flush concurrently (maybe something like a shielded gather), however if this has been tested on 32 partitions and the flush delay isn't a problem then we can table that for a future improvement.

**Pull request closed without merging**